### PR TITLE
Add AQI Sensor + Custom Value color badge

### PR DESCRIPTION
**Pull request by VonOx**
+ Add AQI Sensor + Custom Value color badge
* Reorder DEVICE_FEATURE_CATEGORIES

image
image
image

https://community.gladysassistant.com/t/type-qualite-de-lair-natif-avec-couleurs-en-fonction-de-la-valeur/7014?u=vonox

### CODE REVIEW

File: front/src/components/boxs/device-in-room/device-features/sensor-value/AirQualityDeviceValue.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+  // Hazardous
+  return 'danger';
+};
+
+const BADGE_CATEGORIES = {
+  [DEVICE_FEATURE_CATEGORIES.AIRQUALITY_SENSOR]: value => getBadgeColor(value)
+};
+
+const AirQualityDeviceValue = props => {
```
Comment (atrovato) [inline front/src/components/boxs/device-in-room/device-features/sensor-value/AirQualityDeviceValue.jsx]: Why you don't use BadgeNumberDeviceValue?
You can add you own "colorMethod" for you category.

File: server/utils/constants.js [hunk 2]
```diff
@@ -310,46 +310,47 @@ const INTENTS = {
 };
 
 const DEVICE_FEATURE_CATEGORIES = {
-  LIGHT: 'light',
```
Comment (atrovato) [inline server/utils/constants.js:313]: Did you just sort them?

**Review by atrovato — COMMENTED** [review]
Some comments :)

File: server/utils/constants.js (hunk 2 above)
Comment (VonOx) [reply server/utils/constants.js:313]: Yes https://github.com/VonOx/Gladys/blob/aqi/server/utils/constants.js#L332

File: front/src/components/boxs/device-in-room/device-features/sensor-value/AirQualityDeviceValue.jsx (hunk 1 above)
Comment (VonOx) [reply front/src/components/boxs/device-in-room/device-features/sensor-value/AirQualityDeviceValue.jsx]: I was thinking this feature is too specific and it would be more simple to maintain. I can use BadgeNumber if you prefer.

Comment (Pierre-Gilles) [reply front/src/components/boxs/device-in-room/device-features/sensor-value/AirQualityDeviceValue.jsx]: Agree, this file is exactly like BadgeNumberDeviceValue

You can just use a custom getBadgeColor function specifically for air quality :)

**Review by atrovato — COMMENTED** [review]
Thank you ;)
You can maybe add the feature in demo mode? To get a short preview easily.

**Comment by VonOx** [comment]
@atrovato aqi added into demo file

File: front/src/config/demo.js [hunk 3]
```diff
@@ -263,7 +268,7 @@ const data = {
       }
     ]
   },
-  'get /api/v1/room/living-room?expand=temperature': {
+  'get /api/v1/room/living-room?expand=temperature,devices': {
```
Comment (atrovato) [inline front/src/config/demo.js:271]: Simple question, why did you add "devices" here?

**Review by atrovato — APPROVED** [review]
Nice to me, great job!
Please rebase on master ;)

File: front/src/config/demo.js (hunk 3 above)
Comment (VonOx) [reply front/src/config/demo.js:271]: Browser console complain about this query

**Review by Pierre-Gilles — APPROVED** [review]
Tested all cases locally, and it works fine for me! Thanks for this PR 🙏

Screenshot 2023-02-06 at 12 03 15
Screenshot 2023-02-06 at 12 03 22
Screenshot 2023-02-06 at 12 03 39
Screenshot 2023-02-06 at 12 03 51
Screenshot 2023-02-06 at 12 04 04
Screenshot 2023-02-06 at 12 04 17
